### DISCUSSION TITLE
When publishing crate, add APIs for processing different requests to ensure that crate fails as soon as possible.

### DISCUSSION THREAD

**Discussion by heisen-li** [rust-lang/crates.io#6174] [Feature Requests]
To resolve https://github.com/rust-lang/cargo/issues/8221 and https://github.com/rust-lang/cargo/issues/3662, I plan to send a new validation request before compiling. crates.io seems to be missing the corresponding API. Is it reasonable to add the corresponding API?

**Comment by Turbo87** [comment]
I think the current API might already be sufficient to implement that. 

What is more concerning is that this would lock these API requests to their current form and they would need to be specified in the protocol definition of the communication between cargo and any other registry.

Given that there is a command line option to skip that compile step when publishing I'm not sure it's worth the effort.
